Sources/X/Params/Subspace/ParameterSet.swift

```diff
@@ -5,9 +5,10 @@ public typealias ValueValidatorFunction = (_ value: AnyEncodable) throws -> Void
 // ParamSetPair is used for associating paramsubspace key and field of param
 // structs.
 public struct ParameterSetPair {
-    let key: Data
-    let value: AnyEncodable
-    let validatorFunction: ValueValidatorFunction
+    #warning("TODO JAAP: does this need to be public?")
```

Comment:
I think we should make everything that is not currently accessed elsewhere either private or internal and them expose them when necessary. Some properties will probably be appropriate as public, but I think we can ponder about this later. For now, let's expose the least possible.